TTS/tts/datasets/dataset.py

```diff
@@ -415,7 +415,7 @@ def collate_fn(self, batch):
 
             # get pre-computed d-vectors
             if self.d_vector_mapping is not None:
-                wav_files_names = [batch["wav_file_name"][idx] for idx in ids_sorted_decreasing]
+                wav_files_names = list(batch["wav_file_name"])
```

Comment:
batch is already sorted

Reply:
I this the part that breaks d-vector training ?

Reply:
yeah, `_sort_batch` already returns a sorted batch, when you pick the wav_file_name with the ids from the old order, it shuffles them compared to the rest of the items in the batch.